### ISSUE TITLE
`create_dir_all("")`  should complain about invalid filename

### ISSUE THREAD

**Issue by rosik** [rust-lang/rust#105108]
I tried this code:

```rust
fn main() {
    std::fs::create_dir_all("").unwrap();
}
```

The filename is invalid, so I've expected to see some error similar to posix utils:

```console
$ mkdir -p ""
mkdir: cannot create directory ‘’: No such file or directory
$ mkdir ""
mkdir: cannot create directory ‘’: No such file or directory
```

Instead, nothing interesting happened, `create_dir_all` succeeded.

### Meta

```console
$ rustc --version --verbose
rustc 1.67.0-nightly (e0098a5cc 2022-11-29)
binary: rustc
commit-hash: e0098a5cc3a87d857e597af824d0ce1ed1ad85e0
commit-date: 2022-11-29
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
```

**Comment by rosik** [comment]
P.S. `std::fs::create_dir("")` is fine:

```console
$ cargo run
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/main.rs:2:29
```

I would expect to see the same message from `create_dir_all`

**Comment by ChrisDenton** [comment]
> I don't know why this check exists

For some history see #23383. Though a lot has changed in 7 years.

**Comment by edneedham** [comment]
Thanks for the info, Chris.

**Comment by tbu-** [comment]
Since `Path::new("abc").parent() == Some(Path::new(""))`, it makes sense for `create_dir_all` to succeed in this case. I'd say this needs documentation at most.
